### PR TITLE
cmake: 3.13.4 -> 3.14.5

### PR DESCRIPTION
**Pull request by tobim**
###### Motivation for this change
https://cmake.org/cmake/help/v3.14/release/3.14.html

###### Things done
Besides the updated version and Hash, I applied these changes to the expression:
- Remove `with stdenv.lib` statement to make the origin of various functions clearer at their call sites.
- Calculate `Major.Minor` from `version`, instead of the other way around. Hopefully making this package auto-updater friendly.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by ttuegel — APPROVED** [review]

File: pkgs/development/tools/build-managers/cmake/default.nix [hunk 1]
```diff
@@ -13,24 +13,20 @@
...
-with stdenv.lib;
-
 let
-  os = stdenv.lib.optionalString;
-  majorVersion = "3.13";
-  minorVersion = "4";
-  # from https://cmake.org/files/v3.13/cmake-3.13.4-SHA-256.txt for cmake-3.13.4.tar.gz
-  sha256 = "fdd928fee35f472920071d1c7f1a6a2b72c9b25e04f7a37b409349aef3f20e9b";
-  version = "${majorVersion}.${minorVersion}";
+  lib = stdenv.lib;
```
Comment (JohnAZoidberg) [inline pkgs/development/tools/build-managers/cmake/default.nix]: You can also include `lib` as a parameter in the top-level function.

File: pkgs/development/tools/build-managers/cmake/default.nix [hunk 2]
```diff
@@ -13,24 +13,20 @@
...
+  os = lib.optionalString;
+  version = "3.14.5";
```
Comment (JohnAZoidberg) [inline pkgs/development/tools/build-managers/cmake/default.nix]: `version` and `sha256` don't need to be up here anymore, right?

Comment (tobim) [reply pkgs/development/tools/build-managers/cmake/default.nix]: right. I went ahead and removed the `os` shortcut as well.

**Comment by tobim** [comment]
Rebased on latest staging to fix the conflict.

Should this be backported to 19.03?

**Pull request merged**